substrate/frame/election-provider-multi-block/src/verifier/impls.rs

```diff
@@ -697,15 +684,15 @@ impl<T: Config> Pallet<T> {
 								QueuedSolution::<T>::clear_invalid_and_backings();
 								VerifierWeightsOf::<T>::on_initialize_invalid_terminal()
 							},
-						}
+						};
+
+						weight
 					}
 				},
 				Err(err) => {
 					// the page solution was invalid.
 					Self::deposit_event(Event::<T>::VerificationFailed(current_page, err));
-					StatusStorage::<T>::put(Status::Nothing);
-					QueuedSolution::<T>::clear_invalid_and_backings();
-					T::SolutionDataProvider::report_result(VerificationResult::Rejected);
+					Self::stop();
```

Comment:
tbh I would find this more readable if it was not using a helper function and just do all the steps in here, as it would
be in-line with the rest of the match arms.